core/trino-main/src/main/java/io/trino/sql/planner/DomainTranslator.java

```diff
@@ -911,8 +911,15 @@ yield or(new ComparisonExpression(EQUAL, symbolExpression, coercedLiteral),
 
         private Optional<Object> floorValue(Type fromType, Type toType, Object value)
         {
-            return getSaturatedFloorCastOperator(fromType, toType)
-                    .map(operator -> functionInvoker.invoke(operator, session.toConnectorSession(), value));
+            Optional<ResolvedFunction> operator = getSaturatedFloorCastOperator(fromType, toType);
+            if (operator.isPresent()) {
+                try {
+                    return Optional.of(functionInvoker.invoke(operator.get(), session.toConnectorSession(), value));
+                }
+                catch (TrinoException ignored) {
+                }
```

Comment:
Document why it's necessary and OK to ignore that.

> Add error handling to a coercion
>
> An exception wasn't thrown until today because of rule ordering.

thanks for the info!
can you please enhance the commit message with the info what was guarding us against this?

Reply:
> Document why it's necessary and OK to ignore that.

Done, please see https://github.com/trinodb/trino/pull/16206#discussion_r1155313277 for more details.

> can you please enhance the commit message with the info what was guarding us against this?

Done